### ISSUE TITLE
Addition assignment operator (+=) throws warning but not x = x + y

### ISSUE THREAD

**Issue by stahlmanDesign** [eslint/eslint#8672]
I wrote about the issue in create-react-app and it was suggested to report to ESLint
https://github.com/facebookincubator/create-react-app/issues/2428

To sum it up:
If x is defined and altered using += but not used after that:
`x += y` throws warning `not used` because x is not used after being altered. Fine.
`x = x + y` does NOT throw warning `not used` even though x is never used after being altered

**Comment by viankakrisna** [comment]
If we need to warn if a variable is not used after reassignment, will it also warn on this?
```
let a = 0;
const count = 10
function update(){
   if (a === count){
       return;
   }
    a = a + 1
    update()
}
update()
```
Or eslint can statically analze that update function is called again so `a` is indeed is used after reassignment?

**Comment by not-an-aardvark** [comment]
I think this is a duplicate of https://github.com/eslint/eslint/issues/7578.

**Comment by not-an-aardvark** [comment]
Closing as a duplicate. of #7578.
